test/cb.tests.js

```diff
@@ -0,0 +1,124 @@
+var assert = require('assert'),
+  cb = require('../lib/cb');
+
+function invokeAsync(callback) {
+  setTimeout(function() {
+    callback(null, 'foo');
+  }, 100);
+}
+
+function invokeAsyncError(callback) {
+  setTimeout(function() {
+    callback(new Error());
+  }, 100);
+}
+
+function invokeAsyncTwice(callback) {
+  setTimeout(function() {
+    callback(null, 'foo');
+    callback(null, 'foo');
+  }, 100);
+}
+
+describe('cb(callback)', function() {
+
+  it('should invoke the provided callback', function(done) {
+    invokeAsync(cb(function(err, res) {
+      assert.strictEqual(res, 'foo');
+      done();
+    }));
+  });
+
+  it('shouldn\'t mess with errors', function(done) {
+    invokeAsyncError(cb(function(err, res) {
+      assert(err);
+      done();
+    }));
+  });
+
+  it('should allow multiple executions', function(done) {
+    var count = 0;
+    invokeAsyncTwice(cb(function(err, res) {
+      count++;
+      if (count === 2) done();
+    }));
+  });
+
+});
+
+describe('cb(callback).timeout(ms)', function() {
+	
+  it('should complete successfully within timeout period', function(done) {
+    invokeAsync(cb(function(err, res) {
+      assert.strictEqual(res, 'foo');
+      done();
+    }).timeout(200));
+  });
+
+  it('should complete with an error after timeout period', function(done) {
+    invokeAsync(cb(function(err, res) {
+      assert(err);
```

Comment:
I guess that we should check that the error expected is this one [1] otherwise this could get confused with the `TimeoutError`....

[1] https://github.com/auth0/limitd-redis/pull/34/files#diff-deb070fad931387cd42a3dcf866f39ebc8832e9d0196e80a3ea740563263a1e1R12